autoPyTorch/datasets/base_dataset.py

```diff
@@ -81,16 +149,76 @@ def create_val_split(self,
                     '`val_share` specified, but the Dataset was a given a pre-defined split at initialization already.')
             if val_share < 0 or val_share > 1:
                 raise ValueError(f"`val_share` must be between 0 and 1, got {val_share}.")
-            if holdout_val_type not in self.cross_validators:
+            if not isinstance(holdout_val_type, HoldoutValTypes):
                 raise NotImplementedError(f'The specified `holdout_val_type` "{holdout_val_type}" is not supported.')
             kwargs = {}
             if is_stratified(holdout_val_type):
                 # we need additional information about the data for stratification
                 kwargs["stratify"] = self.train_tensors[-1]
-            train, val = self.holdout_validators[holdout_val_type](val_share, self._get_indices(), **kwargs)
+            train, val = self.holdout_validators[holdout_val_type.name](val_share, self._get_indices(), **kwargs)
             return Subset(self, train), Subset(self, val)
         else:
             if self.val_tensors is None:
                 raise ValueError('Please specify `val_share` or initialize with a validation dataset.')
             val = BaseDataset(self.val_tensors)
             return self, val
+
+    def get_dataset_for_training(self, split: int) -> Tuple[Dataset, Dataset]:
```

Comment:
Maybe something like `split_id` is more descriptive (also since this occurs later in the pipeline as a config option)